diff --git a/chrome/app/framework.exports b/chrome/app/framework.exports
index 37c881f..7044df2 100644
--- a/chrome/app/framework.exports
+++ b/chrome/app/framework.exports
@@ -17,3 +17,11 @@
 # symbols get exported, which will need to be resolved too.
 
+_CCSExtensionList
+_CCSExtensionObserve
+_CCSExtensionInstallResolve
+_CCSExtensionInstallUnpacked
+_CCSExtensionSetEnabled
+_CCSExtensionRemove
+_CCSExtensionSetSiteAccess
+_CCSExtensionPerformAction
 _ChromeAppModeStart_v8
diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 4ecd2c2..c5f7c11 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -3458,4 +3458,6 @@ static_library("ui") {
       "browser_commands_mac.h",
       "browser_commands_mac.mm",
       "browser_mac.cc",
+      "cobble/cobble_extensions.h",
+      "cobble/cobble_extensions.mm",
       "browser_mac.h",
