diff --git a/chrome/browser/ui/browser_web_contents_delegate/browser_web_contents_delegate.cc b/chrome/browser/ui/browser_web_contents_delegate/browser_web_contents_delegate.cc
--- a/chrome/browser/ui/browser_web_contents_delegate/browser_web_contents_delegate.cc
+++ b/chrome/browser/ui/browser_web_contents_delegate/browser_web_contents_delegate.cc
@@ -799,6 +799,10 @@ void BrowserWebContentsDelegate::SetContentsBounds(content::WebContents* source,
 
 void BrowserWebContentsDelegate::UpdateTargetURL(content::WebContents* source,
                                                  const GURL& url) {
+#if BUILDFLAG(IS_MAC)
+  if (cobble_chromium::IsEnabled())
+    cobble_chromium::BrowserTargetURLChanged(source, url);
+#endif
   std::vector<StatusBubble*> status_bubbles =
       browser_ui_controller_->GetStatusBubbles();
   for (StatusBubble* status_bubble : status_bubbles) {
