diff --git a/chrome/browser/extensions/standard_management_policy_provider.cc b/chrome/browser/extensions/standard_management_policy_provider.cc
--- a/chrome/browser/extensions/standard_management_policy_provider.cc
+++ b/chrome/browser/extensions/standard_management_policy_provider.cc
@@ -8,6 +8,7 @@
 
 #include "base/logging.h"
 #include "base/strings/utf_string_conversions.h"
+#include "build/build_config.h"
 #include "chrome/browser/extensions/extension_management.h"
 #include "chrome/grit/generated_resources.h"
 #include "extensions/browser/managed_installation_mode.h"
@@ -18,6 +19,10 @@
 #include "extensions/common/manifest.h"
 #include "extensions/strings/grit/extensions_strings.h"
 #include "ui/base/l10n/l10n_util.h"
+
+#if BUILDFLAG(IS_MAC)
+#include "chrome/browser/ui/cobble/cobble_chromium.h"
+#endif
 
 static_assert(BUILDFLAG(ENABLE_EXTENSIONS_CORE));
 
@@ -107,6 +112,16 @@
   if (Manifest::IsComponentLocation(extension->location())) {
     return true;
   }
+
+#if BUILDFLAG(IS_MAC)
+  if (cobble_chromium::IsEnabled() &&
+      (extension->is_theme() || extension->is_app())) {
+    if (error) {
+      *error = u"Cobble does not support themes or Chrome Apps.";
+    }
+    return false;
+  }
+#endif
 
   // Shared modules are always allowed too: they only contain resources that
   // are used by other extensions. The extension that depends on the shared
